--- a/plugins/meta/portmap/main.go
+++ b/plugins/meta/portmap/main.go
@@ -235,8 +235,7 @@ func parseConfig(stdin []byte, ifName string) (*PortMapConf, *current.Result, er
 				intIdx := *ip.Interface
 				if intIdx >= 0 &&
 					intIdx < len(result.Interfaces) &&
-					(result.Interfaces[intIdx].Name != ifName ||
-						result.Interfaces[intIdx].Sandbox == "") {
+					result.Interfaces[intIdx].Name != ifName {
 					continue
 				}
 			}
